Dockerfile

```diff
@@ -0,0 +1,9 @@
+FROM python:3.7-slim
+WORKDIR /app
+
+COPY ./main.py /app/main.py
+COPY ./setup.py /app/setup.py
+COPY ./README.md /app/README.md
```

Comment:
"In this order"?
How does the *order* matter?

Either way, yes if you edit `main.py` and re-run `docker build -t remote-settings-uptake-health .` again it *will* re-execute everything and do the pip download pip install stuff all over again. 

Does it matter? Is there a better way? ...that is not using volume mounts.

Reply:
> Does it matter? Is there a better way? ...that is not using volume mounts.

Well, it would make sense to not pip download install if setup.py is not changed for example

Reply:
How do you do that? And yes, it would be nice that it runs `pip install ...` if and only if `setup.py` has changed.

Reply:
Well I'm pretty sure `docker build` caches each COPY step based on the file. So putting `main.py` last shoud work

Reply:
I can confirm what I was saying. 

Changing the order in the `Dockerfile` matters... (See `Using cache`)

```
COPY ./setup.py /app/setup.py
COPY ./README.md /app/README.md
RUN pip install -e ".[dev]"
COPY ./main.py /app/main.py
```

```
(.venv) ➜  remote-settings-uptake-worrying git:(0e99dae) ✗ docker build .
[ ... ]
(.venv) ➜  remote-settings-uptake-worrying git:(0e99dae) ✗ vim main.py
(.venv) ➜  remote-settings-uptake-worrying git:(0e99dae) ✗ docker build .
Sending build context to Docker daemon  40.97MB
Step 1/8 : FROM python:3.7-slim
 ---> 96c349a2897f
Step 2/8 : WORKDIR /app
 ---> Using cache
 ---> db33edf62e92
Step 3/8 : COPY ./setup.py /app/setup.py
 ---> Using cache
 ---> 1a9ec710b615
Step 4/8 : COPY ./README.md /app/README.md
 ---> Using cache
 ---> eb0848a67df3
Step 5/8 : RUN pip install -e ".[dev]"
 ---> Using cache
 ---> 9eb77f84b8c8
Step 6/8 : COPY ./main.py /app/main.py
 ---> 986733b4a59a
Step 7/8 : ENTRYPOINT ["python", "main.py"]
 ---> Running in 2be26c4df701
Removing intermediate container 2be26c4df701
 ---> 40616e87c69c
Step 8/8 : CMD []
 ---> Running in 502bd21ddb9d
Removing intermediate container 502bd21ddb9d
 ---> dcd9afa6903d
Successfully built dcd9afa6903d
```

Reply:
You know what caused my confusion? That I thought you meant to change the order amongst the `COPY` commands. Now I totally see what you mean! oh la la.

Reply:
r?